File: app/src/main/java/com/gitsearch/injection/module/ActivityBindingModule.java
package com.gitsearch.injection.module;

import com.gitsearch.ui.main.MainActivity;
import dagger.Module;
import dagger.android.ContributesAndroidInjector;

@Module
public abstract class ActivityBindingModule {

    @ContributesAndroidInjector
    abstract MainActivity mainActivity();

}

File: app/src/main/java/com/gitsearch/injection/module/ActivityBindingModule.kt
package com.gitsearch.injection.module

import com.gitsearch.ui.main.MainActivity
import com.gitsearch.ui.main.MainActivityModule
import dagger.Module
import dagger.android.ContributesAndroidInjector

@Module
abstract class ActivityBindingModule {

    @ContributesAndroidInjector(modules = [MainActivityModule::class])
    abstract fun mainActivity(): MainActivity

}
